dataprofiler/profilers/unstructured_text_profile.py

```diff
@@ -0,0 +1,242 @@
+        
+        merged_profile._case_sensitive = False
+        if self._case_sensitive and other._case_sensitive:
+            merged_profile._case_sensitive = True
+        
+        BaseColumnProfiler._merge_calculations(merged_profile.__calculations,
+                                 self.__calculations,
+                                 other.__calculations)
+
+        if "vocab" in merged_profile.__calculations:
+            merged_profile.vocab = self.vocab.copy()
+            merged_profile._update_vocab(other.vocab)
+            
+        if "words" in merged_profile.__calculations:
+            self._merge_words(other, merged_profile)
+
+        merged_profile.sample_size = self.sample_size + other.sample_size
+
+        return merged_profile
+
+    @property
+    def profile(self):
+        """
+        Property for profile. Returns the profile of the column.
+
+        :return:
+        """
+        word_count = sorted(self.word_count.items(),
+                            key=lambda x: x[1],
+                            reverse=True)
+        profile = dict(
+            vocab=self.vocab,
+            words=list(self.word_count.keys()),
+            word_count=dict(word_count),
+            times=self.times,
+        )
+        return profile
+
+    @BaseColumnProfiler._timeit(name='vocab')
+    def _update_vocab(self, data, prev_dependent_properties=None,
+                      subset_properties=None):
+        """
+        Finds the unique vocabulary used in the text samples.
+
+        :param data: list or array of data from which to extract vocab
+        :type data: Union[list, numpy.array, pandas.DataFrame]
+        :param prev_dependent_properties: Contains all the previous properties
+            that the calculations depend on.
+        :type prev_dependent_properties: dict
+        :param subset_properties: Contains the results of the properties of the
+            subset before they are merged into the main data profile.
+        :type subset_properties: dict
+        :return: None
+        """
+        data_flat = list(itertools.chain(*data))
+        self.vocab = utils._combine_unique_sets(self.vocab, data_flat)
+
+    @BaseColumnProfiler._timeit(name='words')
+    def _update_words(self, data, prev_dependent_properties=None,
+                      subset_properties=None):
+        """
+        Finds the unique words and word count used in the text samples.
+
+        :param data: list or array of data from which to extract vocab
+        :type data: Union[list, numpy.array, pandas.DataFrame]
+        :param prev_dependent_properties: Contains all the previous properties
+            that the calculations depend on.
+        :type prev_dependent_properties: dict
+        :param subset_properties: Contains the results of the properties of the
+            subset before they are merged into the main data profile.
+        :type subset_properties: dict
+        :return: None
+        """
+        if self._case_sensitive:
```

Comment:
less pretty bc somewhat repeated, but faster bc the if isn't computed each loop. I'm okay with it

Reply:
not sure the cost if it were internal, might be miniscule in comparison to lower or findall

Reply:
did some timings, and this actually can result in faster code:
```
if statement in loop: [0.718934227, 0.691763384, 0.7284096360000001, 0.7014499450000002, 0.6951934310000003]
if statement NOT in loop: [0.6208281739999997, 0.6166344240000008, 0.6279374830000002, 0.617970788, 0.6203451770000008]
```
```python
import re
from collections import defaultdict
import timeit


def test_if_speed2(data, is_case_sensitive, stop_words):
    word_count = defaultdict(int)
    if is_case_sensitive:
        for row in data:
            for word in re.findall(r'\w+', row):
                if word.lower() not in stop_words:
                    word_count[word] += 1
    else:
        for row in data:
            for word in re.findall(r'\w+', row):
                word_lower = word.lower()
                if word_lower not in stop_words:
                    word_count[word_lower] += 1


def test_if_speed(data, is_case_sensitive, stop_words):
    word_count = defaultdict(int)
    for row in data:
        for word in re.findall(r'\w+', row):
            word_lower = word.lower()
            if word.lower() not in stop_words:
                if is_case_sensitive:
                    word = word_lower
                word_count[word] += 1
                
    return word_count
            
            
stop_words = {
    'i', 'me', 'my', 'myself', 'we', 'our', 'ours', 'ourselves', 'you',
    "you're", "you've", "you'll", "you'd", 'your', 'yours', 'yourself',
    'yourselves', 'he', 'him', 'his', 'himself', 'she', "she's", 'her',
    'hers', 'herself', 'it', "it's", 'its', 'itself', 'they', 'them',
    'their', 'theirs', 'themselves', 'what', 'which', 'who', 'whom',
    'this', 'that', "that'll", 'these', 'those', 'am', 'is', 'are',
    'was', 'were', 'be', 'been', 'being', 'have', 'has', 'had',
    'having', 'do', 'does', 'did', 'doing', 'a', 'an', 'the', 'and',
    'but', 'if', 'or', 'because', 'as', 'until', 'while', 'of', 'at',
    'by', 'for', 'with', 'about', 'against', 'between', 'into',
    'through', 'during', 'before', 'after', 'above', 'below', 'to',
    'from', 'up', 'down', 'in', 'out', 'on', 'off', 'over', 'under',
    'again', 'further', 'then', 'once', 'here', 'there', 'when',
    'where', 'why', 'how', 'all', 'any', 'both', 'each', 'few', 'more',
    'most', 'other', 'some', 'such', 'no', 'nor', 'not', 'only', 'own',
    'same', 'so', 'than', 'too', 'very', 's', 't', 'can', 'will',
    'just', 'don', "don't", 'should', "should've", 'now', 'd', 'll',
    'm', 'o', 're', 've', 'y', 'ain', 'aren', "aren't", 'couldn',
    "couldn't", 'didn', "didn't", 'doesn', "doesn't", 'hadn', "hadn't",
    'hasn', "hasn't", 'haven', "haven't", 'isn', "isn't", 'ma',
    'mightn', "mightn't", 'mustn', "mustn't", 'needn', "needn't",
    'shan', "shan't", 'shouldn', "shouldn't", 'wasn', "wasn't",
    'weren', "weren't", 'won', "won't", 'wouldn', "wouldn't"
}
data = (['a', 'asdf', 'asdfljds', 'sdlfkjldsa', 'ksldjfkd sldkjf sdf']
        * int(1e2))
            
# if __name__ == '__main__':
print(timeit.repeat(
    'test_if_speed(data, is_case_sensitive=True, stop_words=stop_words)',
    setup='from __main__ import test_if_speed, stop_words, data',
    number=1000,
    repeat=5
))

print(timeit.repeat(
    'test_if_speed2(data, is_case_sensitive=True, stop_words=stop_words)',
    setup='from __main__ import test_if_speed2, stop_words, data',
    number=1000,
    repeat=5
))
```